src/app/backend/secrets.go

```diff
@@ -0,0 +1,114 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package main
+
+import (
+	api "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+// SecretSpec - a single secret representation
+type SecretSpec struct {
+	Name      string            `json:"name"`
+	Namespace string            `json:"namespace,omitempty"`
+	Type      api.SecretType    `json:"type,omitempty"`
+	Data      map[string][]byte `json:"data"`
+}
+
+// SecretsList - response structure for a queried secrets list
+type SecretsList struct {
+	// List of secrets
+	Secrets []SecretSpec `json:"secrets"`
+}
+
+// GetSecrets - return all secrets
+func GetSecrets(client *client.Client) (*SecretsList,
+	error) {
+	namespaces, err := GetNamespaceList(client)
+	if err != nil {
+		return nil, err
+	}
+	// find the secrets from each single known namespace
+	secretsList := &SecretsList{}
+	for _, namespace := range namespaces.Namespaces {
+		secrets, err := client.Secrets(namespace).List(unversioned.ListOptions{
+			LabelSelector: unversioned.LabelSelector{labels.Everything()},
+			FieldSelector: unversioned.FieldSelector{fields.Everything()},
+		})
+		if err != nil {
+			return nil, err
+		}
+		for _, secret := range secrets.Items {
+			secretsList.Secrets = append(secretsList.Secrets, SecretSpec{Name: secret.ObjectMeta.Name, Namespace: secret.ObjectMeta.Namespace, Type: secret.Type})
+		}
+	}
+	return secretsList, err
+}
+
+func CreateSecret(client *client.Client, spec *SecretSpec) (*SecretSpec, error) {
+	namespace := spec.Namespace
+	if &namespace == nil || namespace == "" {
+		namespace = api.NamespaceDefault
+	}
+	secretType := spec.Type
+	if &secretType == nil || secretType == "" {
+		secretType = api.SecretTypeDockercfg
+	}
+	secret := &api.Secret{
+		ObjectMeta: api.ObjectMeta{
+			Name:      spec.Name,
+			Namespace: namespace,
+		},
+		Type: spec.Type,
+		Data: spec.Data,
+	}
+	err := ensureSecretNamespace(client, secret)
+	if err != nil {
+		return nil, err
+	}
+	_, err = client.Secrets(namespace).Create(secret)
+	return &SecretSpec{Name: secret.ObjectMeta.Name, Namespace: secret.ObjectMeta.Namespace, Type: secret.Type, Data: secret.Data}, err
+}
+
+// Checks if the namespace of the given secret exists. Creates it otherwise.
+func ensureSecretNamespace(client *client.Client, secret *api.Secret) error {
```

Comment:
My bad, forgot to remove it. It was needed when the secret is created from file.